src/client/eppo-client.ts

```diff
@@ -32,46 +33,178 @@ export interface IEppoClient {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     subjectAttributes?: Record<string, any>,
     assignmentHooks?: IAssignmentHooks,
-  ): string;
+  ): string | null;
+
+  /**
+   * Maps a subject to a variation for a given experiment.
+   *
+   * @param subjectKey an identifier of the experiment subject, for example a user ID.
+   * @param experimentKey experiment identifier
+   * @param subjectAttributes optional attributes associated with the subject, for example name and email.
+   * The subject attributes are used for evaluating any targeting rules tied to the experiment.
+   * @returns a variation value if the subject is part of the experiment sample, otherwise null
+   * @public
+   */
+  getStringAssignment(
+    subjectKey: string,
+    experimentKey: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    subjectAttributes?: Record<string, any>,
+    assignmentHooks?: IAssignmentHooks,
+  ): string | null;
+
+  getBoolAssignment(
+    subjectKey: string,
+    experimentKey: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    subjectAttributes?: Record<string, any>,
+    assignmentHooks?: IAssignmentHooks,
+  ): boolean | null;
+
+  getNumericAssignment(
+    subjectKey: string,
+    experimentKey: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    subjectAttributes?: Record<string, any>,
+    assignmentHooks?: IAssignmentHooks,
+  ): number | null;
+
+  getJSONAssignment(
+    subjectKey: string,
+    experimentKey: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    subjectAttributes?: Record<string, any>,
+    assignmentHooks?: IAssignmentHooks,
+  ): string | null;
 }
 
 export default class EppoClient implements IEppoClient {
   private queuedEvents: IAssignmentEvent[] = [];
-  private assignmentLogger: IAssignmentLogger = null;
+  private assignmentLogger: IAssignmentLogger | undefined;
 
   constructor(private configurationStore: IConfigurationStore) {}
 
-  getAssignment(
+  public getAssignment(
     subjectKey: string,
     experimentKey: string,
-    subjectAttributes = {},
-    assignmentHooks: IAssignmentHooks = null,
-  ): string {
-    validateNotBlank(subjectKey, 'Invalid argument: subjectKey cannot be blank');
-    validateNotBlank(experimentKey, 'Invalid argument: experimentKey cannot be blank');
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    subjectAttributes: Record<string, any> = {},
+    assignmentHooks?: IAssignmentHooks | undefined,
+  ): string | null {
+    const assignment = this.getAssignmentInternal(
+      subjectKey,
+      experimentKey,
+      subjectAttributes,
+      assignmentHooks,
+      ValueType.StringType,
+    );
+    assignmentHooks?.onPostAssignment(experimentKey, subjectKey, assignment);
 
+    if (assignment !== null)
+      this.logAssignment(experimentKey, assignment, subjectKey, subjectAttributes);
+
```

Comment:
the pre and post assignment events caused a bunch of duplicate code in these handlers